mesonbuild/interpreter.py

```diff
@@ -3019,7 +3019,12 @@ def dependency_impl(self, name, display_name, kwargs):
             # cannot cache them. They must always be evaluated else
             # we won't actually read all the build files.
             if dep.found():
-                self.coredata.deps[identifier] = dep
+                is_cross = self.environment.is_cross_build()
+                if 'native' in kwargs and is_cross:
+                    want_cross = not kwargs['native']
+                else:
+                    want_cross = is_cross
+                self.coredata.deps.put(identifier, want_cross, dep)
```

Comment:
Yeah `self.coredata.deps[for_machine]` would be so much easier.

Reply:
Yeah, now that mention it that probably makes more sense. I'll rework it.